core/vm/eips.go

```diff
@@ -268,6 +268,19 @@ func opMcopy(pc *uint64, interpreter *EVMInterpreter, scope *ScopeContext) ([]by
 	dst := dst.Uint64()
 	length64 := length.Uint64()
 
+  // Note: The spec requires that copying behave "as if" an intermediate
+  // buffer were used. However, actually using an intermediate buffer
+  // raises a DoS vector because it could cause memory allocation even
+  // when no expansion happens. Most memmove implementations check if the
+  // src and dst overlap, and copy backwards if there is overlap (cf.
+  // https://stackoverflow.com/a/22915130/), but does the golang runtime?
+  // `Memory.Set` dispatches into `copy()` builtin, which ultimately calls
+  // runtime.memmove (see https://github.com/golang/go/blob/fd13444b2b4bf1ce6/src/runtime/slice.go#L83).
+  // runtime.memmove itself dispatches into a system-specific optimized
+  // memmove, which is written in assembly and therefore very unlikely to
+  // do any heap allocations - and in fact they indeed do the overlap check.
+  // For instance, see the amd64 implementation:
+  // https://github.com/golang/go/blob/fd13444b2b4bf1ce6/src/runtime/memmove_amd64.s#L78-L82
```

Comment:
The actual golang implementation is irrelevant, but you could instead link to the reference doc: https://pkg.go.dev/builtin#copy 

> The `copy` built-in function copies elements from a source slice into a destination slice. [...] The source and destination may overlap

Reply:
Hm, but actually, you _are_ using a temporary buffer. Or are you? There is no `memory.Get` method, you need to either use `GetPtr` or `GetCopy`. The latter will use a temp buffer, the former not. 
Might be nicer to make this a method on `Memory` instead: like 
`scope.Memory.Copy(dst, src, len uint64)`